Sources/WeatherSensorService/Internal/MockWeatherSensorReader.swift

```diff
@@ -46,14 +46,17 @@ class MockWeatherSensorReader: WeatherSensorReaderType {
         stopSensorReadings()
 
         // Report the first reading immediately, then start the timer
-        reportSensorReadings()
-
-        timer = Timer.scheduledTimer(
-            withTimeInterval: TimeInterval(readerInterval),
-            repeats: true
-        ) { [weak self] _ in
+        DispatchQueue.main.async { [weak self] in
             guard let self = self else { return }
             self.reportSensorReadings()
+            self.timer = Timer.scheduledTimer(
+                withTimeInterval: TimeInterval(self.readerInterval),
+                repeats: true
+            ) { [weak self] _ in
+                guard let self = self else { return }
+                print("reporting")
```

Comment:
Just making sure - do you want to keep this print? That might clutter up the logs.